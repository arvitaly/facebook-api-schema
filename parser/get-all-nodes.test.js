"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
jest.mock("node-fetch");
const node_fetch_1 = require("./../__mocks__/node-fetch");
beforeAll(() => __awaiter(this, void 0, void 0, function* () { return node_fetch_1.start(); }));
afterAll(() => __awaiter(this, void 0, void 0, function* () { return node_fetch_1.stop(); }));
const get_all_nodes_1 = require("./get-all-nodes");
const node_list_1 = require("./node-list");
it("get all nodes", () => __awaiter(this, void 0, void 0, function* () {
    expect(yield get_all_nodes_1.GetAllNodes({
        getNodeList: () => __awaiter(this, void 0, void 0, function* () { return (yield node_list_1.default()).filter((_, i) => i < 2); }),
    })()).toMatchSnapshot();
}));
