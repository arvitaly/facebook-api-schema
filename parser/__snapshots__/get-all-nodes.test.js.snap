// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get all nodes 1`] = `
Array [
  Object {
    "data": Object {
      "description": "Represents a user gaining an achievement in a Facebook game. Not to be confused with an /docs/graph-api/reference/achievement_type/,achievement type.",
      "edges": Array [
        Object {
          "link": "https://developers.facebook.com/docs/graph-api/reference/object/comments/",
          "title": "Comments on the achievement story.",
        },
      ],
      "readingFields": Array [
        Object {
          "description": "ID of this particular achievement.",
          "name": "id",
          "type": Object {
            "object": null,
            "scalar": "string",
          },
        },
        Object {
          "description": "The user who achieved this.",
          "name": "from",
          "type": Object {
            "object": Object {
              "link": "https://developers.facebook.com/docs/graph-api/reference/user/",
              "title": "User",
            },
            "scalar": "User",
          },
        },
        Object {
          "description": "Time at which this was achieved.",
          "name": "publish_time",
          "type": Object {
            "object": null,
            "scalar": "datetime",
          },
        },
        Object {
          "description": "The app in which the user achieved this.",
          "name": "application",
          "type": Object {
            "object": Object {
              "link": "https://developers.facebook.com/docs/graph-api/reference/app/",
              "title": "App",
            },
            "scalar": "App",
          },
        },
        Object {
          "description": "Information about the achievement type this instance is connected with.",
          "name": "data",
          "type": Object {
            "object": null,
            "scalar": "object",
          },
        },
        Object {
          "description": "The achievement type that the user achieved.",
          "name": "achievement",
          "type": Object {
            "object": Object {
              "link": "https://developers.facebook.com/docs/graph-api/reference/achievement_type/",
              "title": "AchievementType",
            },
            "scalar": "AchievementType",
          },
        },
        Object {
          "description": "A weighting given to each achievement type by the app.",
          "name": "importance",
          "type": Object {
            "object": null,
            "scalar": "int",
          },
        },
        Object {
          "description": "Always game.achievement.",
          "name": "type",
          "type": Object {
            "object": null,
            "scalar": "string",
          },
        },
        Object {
          "description": "Indicates whether gaining the achievement published a feed story for the user.",
          "name": "no_feed_story",
          "type": Object {
            "object": null,
            "scalar": "boolean",
          },
        },
      ],
      "readingPermissions": Array [
        "An app access token for any achievements made through that app.",
        "A user access token for any achievements made through the requesting app by that user.",
        "A user access token with user_games_activity permission, for any achievements that person has made in any app. ",
      ],
    },
    "description": "Represents a person gaining a game achievement",
    "link": "https://developers.facebook.com/docs/graph-api/reference/achievement",
    "name": "achievement",
    "title": "Achievement",
  },
  Object {
    "data": Object {
      "description": "A games achievement type created by a Facebook App. Not to be confused with /docs/reference/api/achievement,an instance of an achievement.",
      "edges": Array [],
      "readingFields": Array [],
      "readingPermissions": Array [
        "An app access token is required to view any achievement type created by that app.",
      ],
    },
    "description": "A games achievement type",
    "link": "https://developers.facebook.com/docs/graph-api/reference/achievement_type",
    "name": "achievement_type",
    "title": "Achievement Type",
  },
]
`;
